docs_src/testing_asynchronously/test_main.py

```diff
@@ -0,0 +1,13 @@
+import pytest
+
+from httpx import AsyncClient
+
+import main
+
+
+@pytest.mark.asyncio
+async def test_root():
+    async with AsyncClient(app=main.app, base_url='http://test') as ac:
```

Comment:
You probably need to add `trust_env=False` here as well, otherwise the async client might go through a proxy. Not sure if this is an upstream bug though.

Reply:
My initial thought was to simply follow the [HTTPX example](https://www.python-httpx.org/async/#calling-into-python-web-apps) for calling into python web apps. Also, HTTPX sets [some constraints on the proxies it accepts](https://github.com/encode/httpx/blob/cfd07cc290cc610fceb8d9f025c6bebc1d4b3275/httpx/_utils.py#L287) from the environment, I don't know if that changes anything.

Otherwise:
```suggestion
    async with AsyncClient(app=main.app, base_url='http://test', trust_env=False) as ac:
```

Reply:
https://github.com/encode/httpx/issues/1039 I think it might be a bug.

Reply:
Hm, interesting. What do you propose, we add the `trust_env=False` for now?

Reply:
Yes, I think that's reasonable. Users who have some default network proxy active (e.g. for privacy reasons or in company network situations) will get some bizarre test failures.

Reply:
Hi there, chiming in to confirm that this odd behavior is indeed due to a bug in HTTPX.

So it might be okay to go with this as-is (i.e. without the `trust_env=False` hack), keeping in mind that https://github.com/encode/httpx/issues/1039 looks to be fairly straight-forward to fix (in case anyone wants to give it a go - happy to review a PR, but no expectations of course), so hopefully a fix will be released soon and users equipped with proxies won't encounter this.